api/v1alpha2/workflow_types.go

```diff
@@ -172,8 +173,12 @@ type WorkflowStatus struct {
 	// Indicates whether State has been reached.
 	Ready bool `json:"ready"`
 
-	// User readable reason and status message
-	// +kubebuilder:validation:Enum=Completed;DriverWait;Error
+	// User readable reason and status message.
+	// - Completed: The workflow has reached the state in workflow.Status.State.
+	// - DriverWait: The underlying drivers are currently running.
+	// - TransientCondition: A driver has encountered an error that might not be recoverable.
```

Comment:
Do we want to stress the "might not recover" viewpoint, or the "might recover" viewpoint?
I think I'd like to stress the latter.